### PR TITLE
added mask for lens distortion in images

### PR DESCRIPTION
**Pull request by PxGluz**
This PR adds a mask that accounts for the lens distortion and filters what is shown on the texture. Solves issue #961

### CODE REVIEW

File: libs/MVS/SceneTexture.cpp [hunk 1]
```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
...
 			faceMap.memset((uint8_t)NO_ID);
 		}
+		// TODO: EDIT THIS FUNCTION TO FILTER
 		void Raster(const ImageRef& pt, const Point3f& bary) {
 			const Point3f pbary(PerspectiveCorrectBarycentricCoordinates(bary));
 			const Depth z(ComputeDepth(pbary));
 			ASSERT(z > Depth(0));
 			Depth& depth = depthMap(pt);
+			if (!validFace) {
+				depth = -1;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:171]: as we discussed and tried to explain also on slack, pls do not touch the depth, depth should still record every single face, valid or not so that it do not allow faces in the back to be projected on the image; lets discuss this farther if not clear

File: libs/MVS/SceneTexture.cpp [hunk 2]
```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
...
+		std::vector<ImageRef> prevPixels;
+
 		RasterMesh(const Mesh::VertexArr& _vertices, const Camera& _camera, DepthMap& _depthMap, FaceMap& _faceMap)
-			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap) {}
+			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap), validFace(true) {}
 		void Clear() {
 			Base::Clear();
 			faceMap.memset((uint8_t)NO_ID);
 		}
+		// TODO: EDIT THIS FUNCTION TO FILTER
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:164]: remove

File: libs/MVS/SceneTexture.cpp [hunk 3]
```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
...
 		void Raster(const ImageRef& pt, const Point3f& bary) {
 			const Point3f pbary(PerspectiveCorrectBarycentricCoordinates(bary));
 			const Depth z(ComputeDepth(pbary));
 			ASSERT(z > Depth(0));
 			Depth& depth = depthMap(pt);
+			if (!validFace) {
+				depth = -1;
+				faceMap(pt) = -1;
+				return;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:173]: for same reason, you can not return here and cancel the raster

File: libs/MVS/SceneTexture.cpp [hunk 4]
```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
 		typedef TRasterMesh<RasterMesh> Base;
 		FaceMap& faceMap;
 		FIndex idxFace;
+		cv::Mat mask;
+		bool validFace;
+		std::vector<ImageRef> prevPixels;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:156]: because we have to still handle depth-map rastering, and because there are very few faces invalid, compared to valid once, it is much more expensive to record pixels in this vector, compared to simply raster the face normally and only record if it touched invalid pixels in the mask, and if it did, call the face raster one more time, with a modified version that only sets the faceidx to -1

File: libs/MVS/SceneTexture.cpp [hunk 5]
```diff
@@ -151,24 +151,45 @@ struct MeshTexture {
 		typedef TRasterMesh<RasterMesh> Base;
 		FaceMap& faceMap;
 		FIndex idxFace;
+		cv::Mat mask;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:154]: use out improvements of cv::Mat for images, you will see it is easier to use:
here use Image8u as you only need an image with one unsigned byte channel

File: libs/MVS/SceneTexture.cpp [hunk 6]
```diff
@@ -509,8 +530,35 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		//Creating mask for the image border
+		rasterer.mask = cv::Mat::zeros(imageData.height, imageData.width, CV_8U);
+		cv::Mat imageCopy = imageData.image.clone();
+		cv::cvtColor(imageCopy, imageCopy, cv::COLOR_BGR2GRAY);
+		for (int i = 0; i < imageCopy.rows; i++) {
+			rasterer.mask.at<uint8_t>(i, 0) = imageCopy.at<uint8_t>(i, 0) == 0;
+			rasterer.mask.at<uint8_t>(i, imageCopy.cols - 1) = imageCopy.at<uint8_t>(i, imageCopy.cols - 1) == 0;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:539]: if you use our TImage you will not need to specify again the type here

File: libs/MVS/SceneTexture.cpp [hunk 7]
```diff
@@ -509,8 +530,35 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+					(rasterer.mask.at<uint8_t>(i + 1, j) != 0 || rasterer.mask.at<uint8_t>(i - 1, j) != 0 || rasterer.mask.at<uint8_t>(i, j + 1) != 0 || rasterer.mask.at<uint8_t>(i, j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(i, j) = 255;
+				}
+				if (imageCopy.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) == 0 &&
+					(rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i + 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i - 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j + 1) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) = 255;
+				}
+			}
+		}
+		cv::imwrite(std::to_string(idx) + ".png", rasterer.mask);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:557]: write it to disk only if in verbose mode, check the rest of the code for an example

File: libs/MVS/SceneTexture.cpp [hunk 8]
```diff
@@ -509,8 +530,35 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+				if (imageCopy.at<uint8_t>(i, j) == 0 &&
+					(rasterer.mask.at<uint8_t>(i + 1, j) != 0 || rasterer.mask.at<uint8_t>(i - 1, j) != 0 || rasterer.mask.at<uint8_t>(i, j + 1) != 0 || rasterer.mask.at<uint8_t>(i, j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(i, j) = 255;
+				}
+				if (imageCopy.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) == 0 &&
+					(rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i + 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i - 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j + 1) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) = 255;
+				}
+			}
+		}
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:556]: it is probably a good idea to do a dilate of the mask (see cv::dilate()) to remove pixels at the border of the empty image, due to JPG like artifacts

File: libs/MVS/SceneTexture.cpp [hunk 9]
```diff
@@ -509,8 +530,35 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		//Creating mask for the image border
+		rasterer.mask = cv::Mat::zeros(imageData.height, imageData.width, CV_8U);
+		cv::Mat imageCopy = imageData.image.clone();
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:535]: interesting implementation, but pls try using cv::floodFill() as discussed, which is designed and optimized to do exactly this, plus the code will become easier to read

**Review by cdcseacave — CHANGES_REQUESTED** [review]
cool, pls see comments

File: libs/MVS/SceneTexture.cpp [hunk 10]
```diff
@@ -162,13 +165,20 @@ struct MeshTexture {
 			const Depth z(ComputeDepth(pbary));
 			ASSERT(z > Depth(0));
 			Depth& depth = depthMap(pt);
+			if (mask.at<uint8_t>(pt.y, pt.x) != 0 && validFace)
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:168]: just use `mask(pt.y, pt.x)`

File: libs/MVS/SceneTexture.cpp [hunk 11]
```diff
@@ -162,13 +165,20 @@ struct MeshTexture {
...
+				validFace = false;
 			if (depth == 0 || depth > z) {
 				depth = z;
-				faceMap(pt) = idxFace;
+				if (validFace)
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:172]: you can move the mask check in here

File: libs/MVS/SceneTexture.cpp [hunk 12]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		cv::floodFill(rasterer.mask, cv::Point(0, rasterer.mask.rows / 2), 255);
+		cv::floodFill(rasterer.mask, cv::Point(0, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols / 2, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, rasterer.mask.rows / 2), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, 0), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols / 2, 0), 255);
+
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
+		cv::erode(rasterer.mask, rasterer.mask, kernel);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:536]: sure, erode is fine here as you set the mask for the invalid pixels, instead of the traditional mode of setting it for the valid pixels, either is fine

File: libs/MVS/SceneTexture.cpp [hunk 13]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+
+		cv::threshold(rasterer.mask, rasterer.mask, 254, 255, cv::THRESH_BINARY);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:538]: why is this needed?

File: libs/MVS/SceneTexture.cpp [hunk 14]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+
+
+
+		if (VERBOSITY_LEVEL > 2)
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:542]: cool, this is it

File: libs/MVS/SceneTexture.cpp [hunk 15]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+			cv::imwrite(std::to_string(idx) + ".jpg", rasterer.mask);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:543]: pls add some name too, like invalidmaskXXXX.png
use String::FormatString(), see code for ex

File: libs/MVS/SceneTexture.cpp [hunk 16]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
 		rasterer.Clear();
 		for (auto idxFace : cameraFaces) {
+			rasterer.validFace = true;
 			const Face& facet = faces[idxFace];
 			rasterer.idxFace = idxFace;
 			rasterer.Project(facet);
+			if(!rasterer.validFace)
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:550]: this is perfect

File: libs/MVS/SceneTexture.cpp [hunk 17]
```diff
@@ -151,8 +151,11 @@ struct MeshTexture {
 		typedef TRasterMesh<RasterMesh> Base;
 		FaceMap& faceMap;
 		FIndex idxFace;
+		Image8U mask;
+		bool validFace;
+
 		RasterMesh(const Mesh::VertexArr& _vertices, const Camera& _camera, DepthMap& _depthMap, FaceMap& _faceMap)
-			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap) {}
+			: Base(_vertices, _camera, _depthMap), faceMap(_faceMap), validFace(true) {}
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:158]: no need to intialize validFace here, it will need to be initialized for every raster face

File: libs/MVS/SceneTexture.cpp [hunk 18]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		//Creating mask for the image border
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:522]: pls use the same stile as the rest of the code: comments inside functions start with small letter, and a space between // and word

File: libs/MVS/SceneTexture.cpp [hunk 19]
```diff
@@ -509,11 +519,36 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		cv::floodFill(rasterer.mask, cv::Point(0, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols / 2, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, rasterer.mask.rows - 1), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, rasterer.mask.rows / 2), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols - 1, 0), 255);
+		cv::floodFill(rasterer.mask, cv::Point(rasterer.mask.cols / 2, 0), 255);
+
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
+		cv::erode(rasterer.mask, rasterer.mask, kernel);
+
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:537]: remove space

**Review by cdcseacave — APPROVED** [review]
Looking good now, thx

File: libs/MVS/SceneTexture.cpp (hunk 13 above)
Comment (PxGluz) [reply libs/MVS/SceneTexture.cpp:538]: which line exactly are you referring to? there are 4 lines highlighted in the comment.

Comment (cdcseacave) [reply libs/MVS/SceneTexture.cpp:538]: 538

Comment (PxGluz) [reply libs/MVS/SceneTexture.cpp:538]: it is required to get rid of the black and white version of the original image

File: libs/MVS/SceneTexture.cpp [hunk 20]
```diff
@@ -164,11 +167,18 @@ struct MeshTexture {
 			Depth& depth = depthMap(pt);
 			if (depth == 0 || depth > z) {
 				depth = z;
-				faceMap(pt) = idxFace;
+				if (mask(pt.y, pt.x) != 0 && validFace)
```
[the remark this answers is not in the document]
Comment (cdcseacave) [reply libs/MVS/SceneTexture.cpp:170]: and I would write this three blocks like this:
```
if (validFace && (validFace = (mask(pt.y, pt.x) == 0)))
	faceMap(pt) = idxFace;
else
	faceMap(pt) = -1;
```

File: libs/MVS/SceneTexture.cpp [hunk 21]
```diff
@@ -509,11 +519,31 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		// creating mask for the image border
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:522]: as you update the cv::floodfill usage with mask, pls set the mask valid pixels to 255 and invalid pixels to 0; this will make easier the integration of PR #967

**Review by cdcseacave — APPROVED** [review]

File: libs/MVS/SceneTexture.cpp [hunk 22]
```diff
@@ -509,11 +517,34 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, 0), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, 0), 255);
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
+		cv::erode(rasterer.mask, rasterer.mask, kernel);
+		rasterer.mask = rasterer.mask == 0;
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:536]: why do you invert the values here, when you could simply initialize it with 255 and set 0 in cv::floodfill?

File: libs/MVS/SceneTexture.cpp [hunk 23]
```diff
@@ -509,11 +517,34 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, 0), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, 0), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, 0), 255);
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
+		cv::erode(rasterer.mask, rasterer.mask, kernel);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:535]: using `hiDiff` in cv::floodfill should make the erode unnecessary, correct?

File: libs/MVS/SceneTexture.cpp [hunk 24]
```diff
@@ -509,11 +517,34 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		rasterer.mask = rasterer.mask == 0;
+		if (VERBOSITY_LEVEL > 2) {
+			cv::imwrite(String::FormatString("invalidMask%d.png", idx), rasterer.mask);
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:538]: pls format the index with 4 numbers, with leading 0s
this is required in order to correctly list the files in incremental order

File: libs/MVS/SceneTexture.cpp [hunk 25]
```diff
@@ -509,11 +517,34 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		}
 		rasterer.Clear();
 		for (auto idxFace : cameraFaces) {
+			rasterer.validFace = true;
 			const Face& facet = faces[idxFace];
 			rasterer.idxFace = idxFace;
 			rasterer.Project(facet);
+			if(!rasterer.validFace)
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:546]: always space between if and (

**Review by cdcseacave — APPROVED** [review]

File: libs/MVS/SceneTexture.cpp (hunk 22 above)
Comment (PxGluz) [reply libs/MVS/SceneTexture.cpp:535]: as a parameter convention the mask has to be initialized with 0 so that floodfill will work

Comment (PxGluz) [reply libs/MVS/SceneTexture.cpp:535]: since the mask is a binary matrix it would fill with one the pixels that floodfill changes in the original image, and leave the unchanged ones with 0

Comment (cdcseacave) [reply libs/MVS/SceneTexture.cpp:535]: interesting, I didn't know about all these conventions, thx

File: libs/MVS/SceneTexture.cpp [hunk 26]
```diff
@@ -509,11 +517,33 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
...
+		if (imageCopy(imageCopy.rows - 1, 0) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows - 1), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		if (imageCopy(imageCopy.rows - 1, imageCopy.cols / 2) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, imageCopy.rows - 1), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		if (imageCopy(imageCopy.rows - 1, imageCopy.cols - 1) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows - 1), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		if (imageCopy(imageCopy.rows / 2, imageCopy.cols - 1) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows / 2), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		if (imageCopy(0, imageCopy.cols - 1) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, 0), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		if (imageCopy(0, imageCopy.cols / 2) == 0) cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, 0), 255, &rect, cv::Scalar(0), cv::Scalar(1));
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:534]: this is not needed anymore, right?

**Review by cdcseacave — APPROVED** [review]

**Comment by apurvazaveri** [comment]
Any idea when this shall be approved? I can do some testing of this feature.
